business/health.go

```diff
@@ -87,28 +124,60 @@ func (in *HealthService) getAppHealth(namespace, app, rateInterval string, detai
 func (in *HealthService) GetWorkloadHealth(namespace, workload, rateInterval string) (*models.WorkloadHealth, error) {
 	// Fill all parts
 	health := models.WorkloadHealth{}
-	deployment, err := in.k8s.GetDeployment(namespace, workload)
+	w, err := fetchWorkload(in.k8s, namespace, workload)
 	if err != nil {
 		return nil, err
 	}
-	health.DeploymentStatus = models.DeploymentStatus{
-		Name:              deployment.Name,
-		Replicas:          deployment.Status.Replicas,
-		AvailableReplicas: deployment.Status.AvailableReplicas}
+	health.WorkloadStatus = models.WorkloadStatus{
+		Name:              w.Name,
+		Replicas:          w.Replicas,
+		AvailableReplicas: w.AvailableReplicas}
+
 	health.Requests = in.getWorkloadRequestsHealth(namespace, workload, rateInterval)
 	return &health, nil
 }
 
 // GetNamespaceAppHealth returns a health for all apps in given Namespace (thus, it fetches data from K8S and Prometheus)
 func (in *HealthService) GetNamespaceAppHealth(namespace, rateInterval string) (models.NamespaceAppHealth, error) {
-	appEntities, err := in.k8s.GetNamespaceAppsDetails(namespace)
-	if err != nil {
+	var services []v1.Service
+	var ws models.Workloads
+	cfg := config.Get()
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	errChan := make(chan error, 2)
+
+	go func() {
+		defer wg.Done()
+		var err error
+		services, err = in.k8s.GetServices(namespace, nil)
+		if err != nil {
+			log.Errorf("Error fetching Services per namespace %s: %s", namespace, err)
+			errChan <- err
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		var err error
+		ws, err = fetchWorkloads(in.k8s, namespace, cfg.IstioLabels.AppLabelName)
```

Comment:
So that's going to fetch pods with just `"app"` as selector ... does that work?

Reply:
It should, but I will add an extra test in the app_test.go.
Thanks!

Reply:
I can't add a mocked test as we rely in the k8s label selector:
```
pods, err := in.k8s.CoreV1().Pods(namespace).List(meta_v1.ListOptions{LabelSelector: labelSelector})
```
which has the expected behaviour:
```
$ oc get pods -l app
NAME                          READY     STATUS    RESTARTS   AGE
details-v1                    2/2       Running   0          1h
productpage-v1-1-hpb4b        2/2       Running   0          1h
ratings-v1-7c96469dc4-22tmz   2/2       Running   0          1h
reviews-v1-rg85l              2/2       Running   0          1h
reviews-v2-r9lw4              2/2       Running   0          1h
reviews-v3-0                  2/2       Running   0          1h
$ oc get pods -l bad-app
No resources found.
$ oc get pods -l app=details
NAME         READY     STATUS    RESTARTS   AGE
details-v1   2/2       Running   0          1h
```

So, yes, the Pods fetched to calculate the workloads will be filtered by the "app" label on this line:
```
fetchWorkloads(in.k8s, namespace, cfg.IstioLabels.AppLabelName)
```